src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseWriter.cs

```diff
@@ -66,6 +66,14 @@ public override void Flush()
             _writer.Flush();
         }
 
+        public override Task FlushAsync()
+        {
+            if (IsClosed)
+                ThrowClosed();
```

Comment:
I believe consistency should be our guideline so what we do in other libraries is what we should do (especially System.IO).

From a quick peek in the source code and the previous conversations (I already asked similar question before) I can see what you should do is this:
- if the arguments passed are invalid at the moment of passing (null argument, some negative length etc.) and it will be invalid even when you completely rewrite implementation then you should throw synchronously,
- If there is any slightest chance that it might be invalid only during the runtime (assume implementation may change) then throw asynchronously.

Should you do this in every public sync/async method? Depends: we should consider a risk of doing this change. I believe for the good assessment of compat risk I'd talk to @ericstj.

If the guidelines do not match what we currently have in the code and would make it inconsistent I'd start the design guidelines discussion if we should change everything or follow bad guidelines.